### PR TITLE
Allow subscribing to `RoomInfo` updates

### PR DESCRIPTION
**Pull request by jplatte**
Note to self: Probably should add a changelog entry.

### CODE REVIEW

**Review by Hywan — APPROVED** [review]
It looks good. The idea is to get notified when `RoomInfo` has been updated, but we don't really know which part has been modified. I wonder if that would be desired to avoid unnecessary updates on the client-user-app side? And to clarify: yeah, knowing which part has been modified is difficult.

**Comment by stefanceriu** [comment]
This works well client side but we do get a couple of extra updates whenever we enter a room. I wouldn't expect any updates when that happens 🤔

File: bindings/matrix-sdk-ffi/src/room_info.rs
Comment (Hywan) [inline bindings/matrix-sdk-ffi/src/room_info.rs:49]: Should be removed, see #2421.

**Review by Hywan — CHANGES_REQUESTED** [review]

File: bindings/matrix-sdk-ffi/src/room.rs
Comment (Hywan) [inline bindings/matrix-sdk-ffi/src/room.rs:288]: This code is duplicated from [`matrix_sdk_ui::room_list_service::Room::latest_event`](https://github.com/matrix-org/matrix-rust-sdk/blob/7aa6981e372dac0e3af78a629a34f90e66a24c07/crates/matrix-sdk-ui/src/room_list_service/room.rs#L147-L161). Don't you believe we can refactor to get this code in a single-place?

**Review by Hywan — APPROVED** [review]
Looks good to me. One minor note.

**Comment by Hywan** [comment]
@jplatte Do you think it's just multiple updates, or a bug, in https://github.com/matrix-org/matrix-rust-sdk/pull/2411#issuecomment-1680761307?

**Comment by jplatte** [comment]
Sounds like it *might* be a bug, but hard to tell. Doesn't sound like it's a problem outside of causing a bit of inefficiency though, so we could merge and investigate afterwards, right?

**Comment by jplatte** [comment]
Waiting on feedback from the next test run before merging either way though.

**Comment by stefanceriu** [comment]
> but I'm fine with that.

Yeah, I agree. It's fine. I'm going to try testing it soon, I promise 😁

**Pull request merged**
